### DISCUSSION TITLE
Node.js runtime broken in new release?

### DISCUSSION THREAD

**Discussion by thekiwi** [vercel/vercel#10269] [Help] [answered]
### Problem Description

I've created a new API using the Node.js runtime, it's just the Hello World example for now. But when I run it locally (using `npm start` -- package.json included below) it's throwing me the following error: 
```
502: BAD_GATEWAY
Code: NO_RESPONSE_FROM_FUNCTION
```
with the following log (I've removed personal directory information):
```
Error: Cannot find module 'content-type'
Require stack:
- <removed>/node_modules/@vercel/node/dist/serverless-functions/helpers.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (<removed>/node_modules/@cspotcode/source-map-support/source-map-support.js:811:30)
    at Function.Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (<removed>/node_modules/@vercel/node/dist/serverless-functions/helpers.js:6:24)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Object.require.extensions.<computed> [as .js] (<removed>/node_modules/ts-node/src/index.ts:1608:43)
    at Module.load (node:internal/modules/cjs/loader:1119:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '<removed>/node_modules/@vercel/node/dist/serverless-functions/helpers.js'
  ]
}
```

The code is very simple:

**api/index.ts:**
```TypeScript
import type { VercelRequest, VercelResponse } from '@vercel/node';

export default function (request: VercelRequest, response: VercelResponse) {
  const { name = 'World' } = request.query;
  response.send(`Hello ${name}!`);
}
```
**package.json:**
```JSON
{
  "name": "my-api",
  "version": "1.0.0",
  "description": "",
  "main": "api/index.ts",
  "scripts": {
    "start": "vercel dev",
    "deploy": "vercel deploy --prod"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@types/node": "^20.4.5",
    "@vercel/node": "^2.15.6",
    "typescript": "^5.1.6",
    "vercel": "^31.2.0"
  }
}
```

I downgraded `vercel` to `31.1.1` (the previous release) and this appears to runs fine. But when I compared the two releases on GitHub, the only changes I saw that might be related to this were a large number of dependencies (including `content-type`) being changed from `dev: false` to `dev: true` in the `pnpm-lock.yaml` file.  

I also found [this similar issue](https://github.com/orgs/vercel/discussions/1306) from 6 months ago suggested upgrading `vercel` to fix the same problem, though I'm not sure if the root cause was the same.

**tl;dr** -- I think the most recent `vercel@31.2.0` release might have broken the Node.js runtime wrt. helpers.

**Comment by thekiwi** [comment, marked as answer]
Looks like this has been fixed in PR #10274
